device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -428,14 +435,17 @@ public boolean isEmpty()
      */
     public void registerConnectionStateCallback(IotHubConnectionStateCallback callback, Object callbackContext)
     {
-        /* Codes_SRS_DEVICE_IO_99_001: [The registerConnectionStateCallback shall register the callback with the transport.]*/
         this.transport.registerConnectionStateCallback(callback, callbackContext);
     }
 
-    public void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback statusChangeCallback, Object callbackContext)
+    void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback statusChangeCallback, Object callbackContext, String deviceId)
+    {
+        this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext, deviceId);
+    }
+
+    void registerMultiplexingConnectionStateCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext)
     {
-        //Codes_SRS_DEVICE_IO_34_020: [This function shall register the callback with the transport.]
-        this.transport.registerConnectionStatusChangeCallback(statusChangeCallback, callbackContext);
+        this.transport.registerMultiplexingConnectionStateCallback(callback, callbackContext);
```

Comment:
For my understanding - What is the difference between status callback and mux status callback?

Reply:
Mux callback is fired for connection level events (like TCP connection drop) whereas device specific callbacks are fired for device specific events (like authentication failed to due the key in use being cycled, or if the device isn't registered)